src/main/java/com/artipie/rpm/meta/XmlMetaJoin.java

```diff
@@ -0,0 +1,131 @@
+ *
+ * The above copyright notice and this permission notice shall be included
+ * in all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NON-INFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ */
+package com.artipie.rpm.meta;
+
+import java.io.IOException;
+import java.io.InputStream;
+import java.io.OutputStream;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.StandardCopyOption;
+import javax.xml.stream.XMLEventFactory;
+import javax.xml.stream.XMLEventReader;
+import javax.xml.stream.XMLEventWriter;
+import javax.xml.stream.XMLInputFactory;
+import javax.xml.stream.XMLOutputFactory;
+import javax.xml.stream.XMLStreamException;
+import javax.xml.stream.events.XMLEvent;
+
+/**
+ * Joins two meta xml-files.
+ * @since 0.8
+ */
+public final class XmlMetaJoin {
+
+    /**
+     * Tag.
+     */
+    private final String tag;
+
+    /**
+     * Ctor.
+     * @param tag Metatag
+     */
+    public XmlMetaJoin(final String tag) {
+        this.tag = tag;
+    }
+
+    /**
+     * Appends data from part to target.
+     * @param target Target
+     * @param part File to append
+     * @throws IOException On error
+     */
+    public void merge(final Path target, final Path part) throws IOException {
+        final Path res = target.getParent().resolve(
+            String.format("%s.joined", target.getFileName().toString())
+        );
+        try (OutputStream out = Files.newOutputStream(res)) {
+            final XMLEventWriter writer = XMLOutputFactory.newInstance().createXMLEventWriter(out);
+            this.writeFirstPart(target, writer);
+            this.writeSecondPart(part, writer);
+            writer.close();
+        } catch (final XMLStreamException ex) {
+            throw new IOException(ex);
+        }
+        Files.move(res, target, StandardCopyOption.REPLACE_EXISTING);
+    }
+
+    /**
+     * Writes the first part.
+     * @param target What to write
+     * @param writer Where to write
+     * @throws IOException On error
+     * @throws XMLStreamException On error
+     */
+    private void writeFirstPart(final Path target, final XMLEventWriter writer)
+        throws IOException, XMLStreamException {
+        try (InputStream in = Files.newInputStream(target)) {
+            final XMLEventReader reader = XMLInputFactory.newInstance().createXMLEventReader(in);
+            XMLEvent event;
+            writer.add(reader.nextEvent());
+            writer.add(XMLEventFactory.newFactory().createSpace("\n"));
+            while (reader.hasNext()) {
```

Comment:
@olenagerasimova this can be extracted to a private method to avoid code duplication

Reply:
@Vatavuk hm, I do not see how: `if` statements inside loops are different

Reply:
@olenagerasimova yes, nvm